### PR TITLE
Include symbol for paid products

### PR DESCRIPTION
**Pull request by brunocvcunha**
Still need to go through the list and review more products, but this PR is to get the work started.

### CODE REVIEW

**Comment by brunocvcunha** [comment]
What about the products that supporting companies offers commercial support? Should it get the symbol as well? Or just those clearly have monetization through features on the product?

**Comment by sta-szek** [comment]
I think there could be two symbols:
* for paid products
* for paid products which are free for open-source or small team (like youtrack is free up to 10 users)

**Comment by pacher** [comment]
I would even go further and just mark products which are not open source, regardless of monetization approach.

But that's just my opinion...

File: README.md [hunk 1]
```diff
@@ -307,7 +307,7 @@ A curated list of awesome Java frameworks, libraries and software.
 * [Atomix](http://atomix.io/atomix/) - Fault-tolerant distributed coordination framework.
 * [Axon Framework](http://www.axonframework.org/) - Framework for creating CQRS applications.
 * [Copycat](http://atomix.io/copycat/) - Fault-tolerant state machine replication framework.
-* [Hazelcast](http://hazelcast.org/) - Highly scalable in-memory datagrid.
+* [Hazelcast ![c]](http://hazelcast.org/) - Highly scalable in-memory datagrid.
```
Comment (akullpp) [inline README.md]: Please mention the `free open-source version` of Hazelcast.

**Comment by brunocvcunha** [comment]
@akullpp any thoughts on that?

**Pull request merged**